cumulus/client/service/src/lib.rs

```diff
@@ -755,3 +775,125 @@ impl ParachainInformantMetrics {
 		})
 	}
 }
+
+/// Metrics for the parachain RPC.
+struct ParachainRpcMetrics {
+	/// Time between the submission of a transaction and its inclusion in a backed block.
+	transaction_backed_duration: Histogram,
+}
+
+impl ParachainRpcMetrics {
+	fn new(prometheus_registry: &Registry) -> prometheus::Result<Self> {
+		let transaction_backed_duration = Histogram::with_opts(
+			HistogramOpts::new(
+				"parachain_transaction_backed_duration",
+				"Time between the submission of a transaction and its inclusion in a backed block",
+			)
+			.buckets(linear_buckets(0.01, 40.0, 20).expect("Valid buckets; qed")),
+		)?;
+		prometheus_registry.register(Box::new(transaction_backed_duration.clone()))?;
+
+		Ok(Self { transaction_backed_duration })
+	}
+}
+
+/// Task for monitoring the parachain RPC metrics.
+async fn parachain_rpc_metrics<Block: BlockT>(
+	relay_chain_interface: impl RelayChainInterface + Clone,
+	rpc_transaction_v2_handles: Vec<sc_service::TransactionMonitorHandle<Block::Hash>>,
+	para_id: ParaId,
+	metrics: Option<ParachainRpcMetrics>,
+) -> Result<(), RelayChainError> {
+	const LOG_TARGET: &str = "parachain_rpc_metrics";
+
+	let mut backed_blocks: LruMap<sp_core::H256, ()> = LruMap::new(ByLength::new(64));
+	let mut unresolved_tx: LruMap<sp_core::H256, Vec<Instant>> = LruMap::new(ByLength::new(64));
+
+	/// Convert the substrate block hash to a H256 hash without panics.
+	fn convert_to_h256<Block: BlockT>(
+		hash: &Block::Hash,
+	) -> Result<sp_core::H256, RelayChainError> {
+		if hash.as_ref().len() != 32 {
+			return Err(RelayChainError::GenericError(format!(
+				"Expected hash to be 32 bytes, got {} bytes",
+				hash.as_ref().len()
+			)));
+		}
+
+		Ok(sp_core::H256::from_slice(hash.as_ref()))
+	}
```

Comment:
Why do we need this?

Reply:
I was a bit paranoid about maybe working with different hashes, but don't think that's a real concern :D